File: src/Presentation/A.Presentation/ClientApp/src/app/models/product.ts
import { IBaseEntity } from "./baseEntity";

export interface IProduct extends IBaseEntity {
  id: number;
  name?: string;
  brand?: string;
  type?: string;
  description?: string;
  unitPrice?: number;
  promtionalUnitPrice?: number;
  imageURL?: string;
  promotion?: string;
  promotionDescription?: string;
  stockQuantity: number;
  basketQuantity: number;
}

export interface IProductBrand {
  id: number;
  brand: string;
}

export interface IProductType {
  id: number;
  type: string;
}

File: src/ApiGateway/A.APIGateway/Models/Product.cs
using System.ComponentModel.DataAnnotations;

namespace A.APIGateway.Models
{
    public class Product : BaseEntity
    {
        public string? Name { get; set; }
        public string? Brand { get; set; }
        public string? Type { get; set; }
        public string? Description { get; set; }

        public double UnitPrice { get; set; }
        public double PromtionalUnitPrice { get; set; } = 0;

        public string? ImageURL { get; set; }

        public string? Promotion { get; set; }
        public string? PromotionDescription { get; set; }

        public int StockQuantity { get; set; } = 0;
        public int BasketQuantity { get; set; } = 0;
    }
}
